### PR TITLE
MCPClient: refactor archivematicaClient.py

### PR DESCRIPTION
**Pull request by jrwdunham**
Adds docstrings, more Pythonic formatting, and some cosmetic structural
changes to archivematicaClient.py

### CODE REVIEW

File: src/MCPClient/lib/archivematicaClient.py [hunk 1]
```diff
@@ -1,8 +1,34 @@
 #!/usr/bin/env python2
 
+"""Archivematica (Gearman) Client
```
Comment (sevein) [inline src/MCPClient/lib/archivematicaClient.py]: To be accurate, it's a Gearman worker.

File: src/MCPClient/lib/archivematicaClient.py [hunk 2]
```diff
@@ -21,166 +47,190 @@
...
+        script, task_uuid, capture_output = _process_gearman_job(
+            gearman_job, gearman_worker)
+    except ProcessGearmanJobError as error:
+        return cPickle.dumps(error)
     except OSError:
         logger.exception('Execution failed')
-        output = ["Archivematica Client Error!", traceback.format_exc()]
-        exitCode = 1
-        return cPickle.dumps({"exitCode": exitCode, "stdOut": output[0], "stdError": output[1]})
+        return cPickle.dumps({'exitCode': 1,
```
Comment (sevein) [inline src/MCPClient/lib/archivematicaClient.py:190]: ^ this `OSError` was used to control the exceptions raised by subprocess (executeOrRun), should be done inside _process_gearman_job instead?

**Review by sevein — CHANGES_REQUESTED** [review]
I have a couple of small observations but looking very good! It should target qa/1.x, this is definitely something we want to include!

**Comment by jrwdunham** [comment]
Replaced by https://github.com/artefactual/archivematica/pull/829

**Pull request closed without merging**